src/solvers/flattening/boolbv_extractbit.cpp

```diff
@@ -19,11 +19,13 @@ Author: Daniel Kroening, kroening@kroening.com
 literalt boolbvt::convert_extractbit(const extractbit_exprt &expr)
 {
   const bvt &src_bv = convert_bv(expr.src());
+  const auto &index = expr.index();
 
   // constant?
-  if(expr.index().is_constant())
+  if(index.is_constant())
```

Comment:
I think more idiomatic in this code base to:

```C++
if(const auto &constant_index = try_expr_dynamic_cast<constant_exprt>(index))
{
...
  numeric_cast_v<mp_integer>(constant_index);
```
Since seems plausible the `is_constant` method will go away at some point

Reply:
or shorter using an optional:
```c
if(const auto index_as_integer = numeric_cast<mp_integer>(index))
```